sdk/dapp-kit/src/components/SuiClientProvider.tsx

```diff
@@ -40,31 +49,39 @@ const DEFAULT_CREATE_CLIENT = function createClient(
 };
 
 export function SuiClientProvider<T extends NetworkConfigs>(props: SuiClientProviderProps<T>) {
+	const { onNetworkChange, network, children } = props;
 	const networks = (props.networks ?? DEFAULT_NETWORKS) as T;
 	const createClient =
 		(props.createClient as typeof DEFAULT_CREATE_CLIENT) ?? DEFAULT_CREATE_CLIENT;
 
 	const [selectedNetwork, setSelectedNetwork] = useState<keyof T & string>(
-		props.defaultNetwork ?? (Object.keys(networks)[0] as keyof T & string),
+		props.network ?? props.defaultNetwork ?? (Object.keys(networks)[0] as keyof T & string),
 	);
 
-	const [client, setClient] = useState<SuiClient>(() => {
-		return createClient(selectedNetwork, networks[selectedNetwork]);
-	});
+	const currentNetwork = props.network ?? selectedNetwork;
+
+	const client = useMemo(() => {
+		return createClient(currentNetwork, networks[currentNetwork]);
+	}, [createClient, currentNetwork, networks]);
 
 	const ctx = useMemo((): SuiClientProviderContext => {
 		return {
 			client,
 			networks,
 			selectedNetwork,
-			selectNetwork: (network) => {
-				if (network !== selectedNetwork) {
-					setSelectedNetwork(network);
-					setClient(createClient(network, networks[network]));
+			selectNetwork: (newNetwork) => {
+				if (currentNetwork === newNetwork) {
+					return;
 				}
+
+				if (!network && newNetwork !== selectedNetwork) {
+					setSelectedNetwork(newNetwork);
+				}
+
+				onNetworkChange?.(newNetwork);
```

Comment:
should this only be called if the network is actually changes?

Reply:
there's an early return above to prevent that

Reply:
oh sorry oops, I mis-read the `newNetwork !== selectedNetwork` check above